crates/rattler_virtual_packages/src/cuda.rs

```diff
@@ -26,10 +27,14 @@ pub fn cuda_version() -> Option<Version> {
         .clone()
 }
 
+const CUDA_OVERRIDE_ENV: &str = "CONDA_OVERRIDE_CUDA";
 /// Attempts to detect the version of CUDA present in the current operating system by employing the
 /// best technique available for the current environment.
 pub fn detect_cuda_version() -> Option<Version> {
```

Comment:
This function should be pure functional without environmental side-effects.

Instead I propose we add some sort of `from_env` function to the appropriate virtual packages that allow users to choose whether they want this behavior. Or a `or_override_with_env_var` or something along those lines.

We should also debate the environment variable name. I believe conda has a different name? Im also not sure whether we should hardcode that in rattler itself.